pkg/intoto/metadata.go

```diff
@@ -9,89 +9,148 @@ import (
 	canonicaljson "github.com/docker/go/canonical/json"
 )
 
-// Metadata represents the In-Toto metadata stored in TUF.
-// All fields are represented as []byte in order to be stored in the Custom field for TUF metadata.
+// Metadata points to root layout, its public keys, and/or links
 type Metadata struct {
-	// TODO: remove this once the TUF targets key is used to sign the root layout
-	Key    []byte            `json:"key"`
-	Layout []byte            `json:"layout"`
-	Links  map[string][]byte `json:"links"`
+	Data       []byte   `json:"data"`
+	PublicKeys []string `json:"public-keys"` // filenames
+	Links      []string `json:"links"`       // filenames
 }
 
-// WriteMetadataFiles writes the content of a metadata object into files in a directory
-func WriteMetadataFiles(m *Metadata, dir string) error {
-	abs, err := filepath.Abs(dir)
-	if err != nil {
-		return err
-	}
+// Custom is a generic structure that contains in-toto Metadata
+type Custom struct {
+	InToto Metadata `json:"in-toto"`
```

Comment:
Not 100% sure about the JSON keys using a dash.
While there is no official convention for keys ([here's the official JSON standard](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf)), I have mostly seen `camelCase` (most often), or `snake_case`.

For reference, [CNAB Core](https://github.com/cnabio/cnab-spec/blob/master/101-bundle-json.md) uses `camelCase`, while [TUF](https://github.com/theupdateframework/specification/blob/master/tuf-spec.md) uses `snake_case`.

Personally, given that the custom metadata is eventually added to `targets.json`, I would prefer it followed the TUF naming convention (for example `consistent_snapshot`).

What do you think?

Reply:
I know realize this may have been a mistake on my part, as the spec already shows the `"in-toto"` keys in the examples, which is the key I initially used when bootstrapping this - https://github.com/cnabio/cnab-spec/blob/master/301-metadata-repositories.md

I apologize for the confusion (and for potentially selecting a rather unfortunate JSON key name..) - I will revert the implementation here, unless you have ideas here?

Reply:
Sounds good to me. Shouldn't be a problem unless we use something like an ORM that translates keys into variable names...